### DISCUSSION TITLE
RawBlock not found by pandoc depending on how file is passed (tex to html)

### DISCUSSION THREAD

**Discussion by Pecnut** [jgm/pandoc#10270] [Q&A] [answered]
Hello Pandoc team!

I am following the example of [Building images with TikZ](https://pandoc.org/lua-filters.html#building-images-with-tikz). 

Why does this command work (exactly as given in the example):

```
pandoc --lua-filter tikz.lua -s -o cycle.html <<EOF
Here is a diagram of the cycle:

\begin{tikzpicture}

\def \n {5}
\def \radius {3cm}
\def \margin {8} % margin in angles, depends on the radius

\foreach \s in {1,...,\n}
{
  \node[draw, circle] at ({360/\n * (\s - 1)}:\radius) {$\s$};
  \draw[->, >=latex] ({360/\n * (\s - 1)+\margin}:\radius)
    arc ({360/\n * (\s - 1)+\margin}:{360/\n * (\s)-\margin}:\radius);
}
\end{tikzpicture}
EOF
```

(and in so-doing, converts the tikz into an SVG file and correctly generates the HTML page)

but if I place the LaTeX code into a separate file, say `cycle.tex`, and run

`pandoc --lua-filter tikz.lua -s -o cycle.html cycle.tex`

then the output file is still created, but the image isn't converted at all?

Some digging suggests that Pandoc isn't interpreting the \begin{tikzpicture} \end{tikzpicture} environment at all: as far as I can tell from looking at what I think is the document tree, this is being ignored rather than being parsed as a RawBlock, whereas when I give the tex file using <<EOF, pandoc *does* parse it as a RawBlock within the document tree.

I am so confused. Please help!

(I am running Pandoc 3.5 on a Mac)

**Comment by tarleb** [comment, marked as answer]
Pandoc tries to guess the input format based of the filename extension, so it uses the `latex` reader in your second case above. If there is no input filename, as in the first case, then pandoc defaults to `markdown`.

The LaTeX reader skips TeX commands that it cannot handle, while the Markdown reader puts all LaTeX blocks in the `RawBlock` elements that are then processed by the filter. You could try to set `--from=markdown` or `--from=latex+raw_tex` when using an input file with a `.tex` extension. Enabling the `+raw_tex` extension on the LaTeX reader ensures that unknown TeX blocks are kept around instead of just being skipped.

**Comment by Pecnut** [comment]
This is super helpful, thank you @tarleb ! The example above now works and I feel enlightened. 👍 

This somewhat annoyingly runs into a problem with the larger project I'm working on, where `--from=latex` correctly parses `\subfile` (from the `subfiles` LaTeX package), but `--from=latex+raw_tex` appears to ignore the `\subfile` command (and so doesn't include the contents of the named subfile).

Any idea how to get it to work? Obviously `--from=latex+raw_tex` is what I need to get the image conversion working, but it doesn't appear to interpet `\subfile` in the way that `--from=latex` does. At the moment it seems I get one or the other! 🫤 

(If you think this is better asked as a separate question, I am happy to do that!)

**Reply by tarleb** [reply]
I'm afraid you'll need some kind of filter to get it to work.

**Comment by Pecnut** [comment]
OK that's great, thank you @tarleb !
